src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -628,6 +628,16 @@ internal int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancella
                         }
                     }
 
+                    string finalRefPeFilePath = Arguments.OutputRefFilePath;
+                    if (finalRefPeFilePath != null)
+                    {
+                        if (!CompileAndEmitRefAssembly(compilation, emitOptions.WithEmitMetadataOnly(true),
+                            finalRefPeFilePath, touchedFilesLogger, diagnosticBag, cancellationToken))
+                        {
+                            return Failed;
```

Comment:
Maybe we should still write touched files (see below) even if some part this failed? What do you think?